src/test/regress/expected/local_shard_execution.out

```diff
@@ -2432,6 +2443,392 @@ DEBUG:  Creating router plan
+-- coordinated_transaction_should_use_2PC prints the internal
+-- state for 2PC decision on Citus. However, even if 2PC is decided,
+-- we may not necessarily use 2PC over a connection unless it does
+-- a modification
+CREATE OR REPLACE FUNCTION coordinated_transaction_should_use_2PC()
+RETURNS BOOL LANGUAGE C STRICT VOLATILE AS 'citus',
+$$coordinated_transaction_should_use_2PC$$;
+-- make tests consistent
+SET citus.max_adaptive_executor_pool_size TO 1;
+RESET citus.enable_metadata_sync;
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions
+---------------------------------------------------------------------
+                             0
+(1 row)
+
+-- single shard local command without transaction block does set the
+-- internal state for 2PC, but does not require any actual entries
+WITH cte_1 AS (INSERT INTO event_responses_no_pkey VALUES (2, 2, 'yes') RETURNING *)
+SELECT coordinated_transaction_should_use_2PC() FROM cte_1;
+DEBUG:  only SELECT, UPDATE, or DELETE common table expressions may be router planned
+DEBUG:  generating subplan XXX_1 for CTE cte_1: INSERT INTO public.event_responses_no_pkey (event_id, user_id, response) VALUES (2, 2, 'yes'::public.invite_resp) RETURNING event_id, user_id, response
+DEBUG:  Creating router plan
+DEBUG:  query has a single distribution column value: 2
+DEBUG:  Plan XXX query after replacing subqueries and CTEs: SELECT public.coordinated_transaction_should_use_2pc() AS coordinated_transaction_should_use_2pc FROM (SELECT intermediate_result.event_id, intermediate_result.user_id, intermediate_result.response FROM read_intermediate_result('XXX_1'::text, 'binary'::citus_copy_format) intermediate_result(event_id integer, user_id integer, response public.invite_resp)) cte_1
+DEBUG:  Creating router plan
+NOTICE:  executing the command locally: INSERT INTO public.event_responses_no_pkey_1480007 (event_id, user_id, response) VALUES (2, 2, 'yes'::public.invite_resp) RETURNING event_id, user_id, response
+NOTICE:  executing the command locally: SELECT public.coordinated_transaction_should_use_2pc() AS coordinated_transaction_should_use_2pc FROM (SELECT intermediate_result.event_id, intermediate_result.user_id, intermediate_result.response FROM read_intermediate_result('XXX_1'::text, 'binary'::citus_copy_format) intermediate_result(event_id integer, user_id integer, response public.invite_resp)) cte_1
+ coordinated_transaction_should_use_2pc
+---------------------------------------------------------------------
+ t
+(1 row)
+
+SELECT count(*) FROM pg_dist_transaction;
+ count
+---------------------------------------------------------------------
+     0
+(1 row)
+
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions
+---------------------------------------------------------------------
+                             0
+(1 row)
+
+-- single shard local modification followed by a single shard
+-- not-local modification uses 2PC
+BEGIN;
```

Comment:
This test and some others makes sure that [`Activate2PCIfModifyingTransactionExpandsToNewNode()`](https://github.com/citusdata/citus/blob/0944c631f1eaa5b51478631ab40c1b4ad57d2cd2/src/backend/distributed/executor/adaptive_executor.c#L3832) covers when a local execution TX expands with a modification command to a remote node. In that case, the second command activates 2PC as expected